### ISSUE TITLE
🐛 Pug resolves component tag wrongly in Vue files

### ISSUE THREAD

**Issue by Hammster** [parcel-bundler/parcel#1089]
To load dynamic components into a vue component, the component tag is used. Those tags are handled as singluar tags `<component id="componentName"/>` or closed tags  `<component id="componentName"></component>`

If i use pug to define the template of my component, it sometimes transpiles it correctly and sometimes the tag is not closed at all e.g `<component id="componentName">`

### 🤔 Expected Behavior

``` pug
.myclass
  component.myspecialcomponent(is="fancyName")
```

transpiles to 

``` html
<div class="myclass">
  <component is="fancyName" class="myspecialcomponent"></component>
</div>
```

<div

### 😯 Current Behavior

Sometimes the Expected Behavior and sometimes this

``` html
<div class="myclass">
  <component is="fancyName" class="myspecialcomponent">
</div>
```

### 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | master
| Node             | 8.2.1
| npm/Yarn         | 1.5.1
| Operating System | Win 10

**Comment by Hammster** [comment]
I narrowed the error down, it happens if the component has a svg use inside of it that uses a `xlink:href`

Example
``` pug
.pane
  .pane-tabs
    ui-button(v-for="paneContent in paneContentList" @click="setPaneContent(paneContent)") {{paneContent}}
  .pane-content
    component.test(:is="paneContentVisible" v-if="!paneContentVisible.includes('ui-')")
    component(:is="paneContentVisible" 
      v-if="paneContentVisible === 'ui-fuzzysearch'",
      v-bind:items="['alpha', 'alpy', 'alpha omega','beta delta','gamma sigma','epsylon zeta']", 
      v-bind:preselected="1"
    )
    component(:is="paneContentVisible" v-if="paneContentVisible === 'ui-button'") Teststring
    svg
      use(xlink:href="../../svg/symbols.svg#ei-close")
```

**Comment by DeMoorJasper** [comment]
Does this happen in watch or build mode?

**Comment by Hammster** [comment]
Same issue on node 9.10.0 also i use yarn@1.5.1 which is the latest

I have a build script that uses the parcel API, basicall i run node build/parcel.js

``` javascript
const Bundler = require('parcel-bundler')
const path = require('path')
const chalk = require('chalk')
// Entrypoint file location
const file = {
  electron: path.join(__dirname, '../src/electron/index.js'),
  renderer: path.join(__dirname, '../src/renderer/index.html')
}

// Bundler options
const options = {
  electron : {
    outDir: './dist/electron', // The out directory to put the build files in, defaults to dist
    outFile: 'index.js', // The name of the outputFile
    publicUrl: './',
    watch: true, // whether to watch the files and rebuild them on change, defaults to process.env.NODE_ENV !== 'production'
    cache: false, // Enabled or disables caching, defaults to true
    minify: false, // Minify files, enabled if process.env.NODE_ENV === 'production'
    target: 'electron', // browser/node/electron, defaults to browser
    https: false, // Server files over https or http, defaults to false
    logLevel: 3, // 3 = log everything, 2 = log warnings & errors, 1 = log errors
    hmrPort: 0, // The port the hmr socket runs on, defaults to a random free port (0 in node.js resolves to a random free port)
    sourceMaps: true, // Enable or disable sourcemaps, defaults to enabled (not supported in minified builds yet)
    hmr: true,
    hmrHostname: '', // A hostname for hot module reload, default to ''
    detailedReport: false // Prints a detailed report of the bundles, assets, filesizes and times, defaults to false, reports are only printed if watch is disabled
  },
  renderer: {
    outDir: './dist/renderer', // The out directory to put the build files in, defaults to dist
    outFile: 'index.html', // The name of the outputFile
    publicUrl: './',
    watch: true, // whether to watch the files and rebuild them on change, defaults to process.env.NODE_ENV !== 'production'
    cache: false, // Enabled or disables caching, defaults to true
    minify: false, // Minify files, enabled if process.env.NODE_ENV === 'production'
    target: 'electron', // browser/node/electron, defaults to browser
    logLevel: 3, // 3 = log everything, 2 = log warnings & errors, 1 = log errors
    hmrPort: 0, // The port the hmr socket runs on, defaults to a random free port (0 in node.js resolves to a random free port)
    sourceMaps: true, // Enable or disable sourcemaps, defaults to enabled (not supported in minified builds yet)
    hmr: true,
    hmrHostname: '', // A hostname for hot module reload, default to ''
    detailedReport: false // Prints a detailed report of the bundles, assets, filesizes and times, defaults to false, reports are only printed if watch is disabled
  }
}

// Initialises a bundler using the entrypoint location and options provided
const bundler = {
  electron: new Bundler(file.electron, options.electron),
  renderer: new Bundler(file.renderer, options.renderer)
}

async function resolveBundles () {
  const resolvedBundles = {}
  for (const bundleName of ['electron', 'renderer']) {
    // Awaiting one bundle after another
    console.log(chalk.cyan.bold('■  Bundel: ') + bundleName)
      console.log('')
    }
    resolvedBundles[bundleName] = await bundler[bundleName].bundle()
  }
}

// Run the bundler, this returns the main bundle
// Use the events if you're using watch mode as this promise will only trigger once and not for every rebuild
resolveBundles()


```

This is the precise error the cli is outputting

```
×  C:\Workspace\Secretproject\Frontend\src\renderer\components\script-view\index.vue: tag <component> has no matching end tag.
    at VueAsset.compileTemplate (C:\Workspace\Secretproject\Frontend\node_modules\parcel-bundler\src\assets\VueAsset.js:131:15)
    at VueAsset.postProcess (C:\Workspace\Secretproject\Frontend\node_modules\parcel-bundler\src\assets\VueAsset.js:77:26)
    at Pipeline.processAsset (C:\Workspace\Secretproject\Frontend\node_modules\parcel-bundler\src\Pipeline.js:70:31)
    at <anonymous>
```

**Comment by Hammster** [comment]
Okay so i've found the issue but i don't know how to handle it properly.

`posthtml-renderer` filters the component tag out as a a single tag because posthtml uses it. This does not apply to vue's component tag which can hold content.

here the code from posthtml-renderer

https://github.com/posthtml/posthtml-render/blob/cbd69ea709034a18eb64749407322a6199e588e5/lib/index.js (edited)

The parsing to posthtml-renderer happens in the HTMLAsset.js on line 164 when the ast is dirty. This happens if a `use` with the `xlink:href` attribute is present.

**Comment by Hammster** [comment]
Issue will be fixed in post-html@1.1.3

**Comment by Hammster** [comment]
I made a pullrequest that sets `post-html@1.1.3` as the minimal version
